### ISSUE TITLE
anytime function changing the input variable

### ISSUE THREAD

**Issue by ahstat** [eddelbuettel/anytime#61]
Hi,
The function anytime(x) will change the object x when applied, in the following example (anytime 0.2.2 with R 3.3.3). Is this the normal behavior of the package?

```R
library(anytime)

# as.numeric(as.POSIXct("2017-01-01 12:34:56", tz = "Asia/Shanghai"))
unix_time1 = 1483245296 
unix_time2 = unix_time1
unix_time3 = unix_time1+1-1

time1 = anytime(unix_time1)
print(time1)
print(unix_time1)
print(unix_time2)
print(unix_time3)
```

My output is (respectively): 
"2017-01-01 12:34:56 CST" 
"2017-01-01 12:34:56 CST" #I expected 1483245296
"2017-01-01 12:34:56 CST" #I expected 1483245296
1483245296

**Comment by eddelbuettel** [comment]
Yes, I think that is normal for R. It's also specific to the numeric input.

**Comment by eddelbuettel** [comment]
Also consider starting with `unix_time1 <- 1483245296L` which makes it an `integer`.  Now a copy has to be made to create a new type -- `numeric` as required for a `POSIXct` -- and the behaviour is more similar to all other cases (ie parsing from text).

What you see here is a side effect of you supplying a `numeric` on input, which simple gets recast to another type _but not copied_.

**Comment by ahstat** [comment]
Thank you for the answer, I understand the behavior now. It is a little different from the classic copy-on-modify behavior of most R functions (I found out this post: http://stackoverflow.com/questions/15759117/what-exactly-is-copy-on-modify-semantics-in-r-and-where-is-the-canonical-source ).

**Comment by eddelbuettel** [comment]
Thanks for circling back.  I also realized since we first chatted here that it is related to copy-on-write but in really weird way:  we are actually _not_ writing (as in: altering) the value, we just set a class attribute tag.

Conceptually, one could potentially desire to get a copy in that case too.  I guess the current behaviour is more lightweight "and the way it has been done before" so I would not expect change.

But it made for a very interesting head-scratcher!
